restli-server/src/main/java/com/linkedin/restli/internal/server/response/ResponseUtils.java

```diff
@@ -97,7 +117,19 @@ public static RestResponse buildResponse(RoutingResult routingResult, RestLiResp
                                                          .getResourceEntityType();
     if (restLiResponse.hasData() && ResourceEntityType.STRUCTURED_DATA == resourceEntityType)
     {
+      DataSchema dataSchema = null;
+      Class<? extends RecordTemplate> valueClass = routingResult.getResourceMethod().getResourceModel().getValueClass();
+      if (valueClass != null)
+      {
+        dataSchema = DataTemplateUtil.getSchema(valueClass);
+      }
```

Comment:
move this inside if statement if we have fill defaults

Reply:
+1, logic is only necessary when needed

Reply:
Updated